all-actors/src/main/java/org/sunbird/NotificationValidator.java

```diff
@@ -0,0 +1,81 @@
+package org.sunbird;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.sunbird.pojo.NotificationMode;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.message.IResponseMessage;
+import org.sunbird.message.ResponseCode;
+import org.sunbird.pojo.NotificationRequest;
+import org.sunbird.request.Request;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Validates send notification api request
+ */
+public class NotificationValidator {
+
+
+    public static void validateSendNotificationRequest(Request request) throws BaseException {
+        List<NotificationRequest> req;
+        try {
+            req = NotificationRequestMapper.toList((List<Map<String, Object>>) request.getRequest().get(JsonKey.NOTIFICATIONS));
+        } catch (Exception e) {
+            throw new BaseException("INVALID_REQUESTED_DATA",
+                    MessageFormat.format(IResponseMessage.INVALID_REQUESTED_DATA, ", provide a valid request data"),
+                    ResponseCode.CLIENT_ERROR.getCode());
+        }
+        if (req.isEmpty()) {
+            throw new BaseException("MANDATORY_PARAMETER_MISSING",
+                    MessageFormat.format(IResponseMessage.MANDATORY_PARAMETER_MISSING, JsonKey.NOTIFICATIONS),
+                    ResponseCode.CLIENT_ERROR.getCode());
+        }
+        for (NotificationRequest notificationRequest : req) {
+            validateSendNotificationModeType(notificationRequest.getMode());
```

Comment:
have one high level 'validate' function - please encapsulate these lower details within it. Also, name simple - validateModeType, validateIds, validateParms. In fact, I'd ask what's the diff between validateMandatoryParams and validateModeType - why can't one do the other as well. Just think.

Reply:
I changed the method names, but I am not getting this  ("In fact, I'd ask what's the diff between validateMandatoryParams and validateModeType - why can't one do the other as well. Just think."